stl/inc/bit

```diff
@@ -34,6 +34,49 @@ _NODISCARD constexpr _To bit_cast(const _From& _Val) noexcept {
     return __builtin_bit_cast(_To, _Val);
 }
 
+#if _HAS_CXX23
+_NODISCARD constexpr unsigned short _Byteswap_ushort(const unsigned short _Val) noexcept {
+    if (_STD is_constant_evaluated()) {
+        return static_cast<unsigned short>((_Val << 8) | (_Val >> 8));
+    } else {
+        return _byteswap_ushort(_Val);
+    }
+}
+
+_NODISCARD constexpr unsigned long _Byteswap_ulong(const unsigned long _Val) noexcept {
+    if (_STD is_constant_evaluated()) {
+        return (_Val << 24) | ((_Val << 8) & 0x00FF'0000) | ((_Val >> 8) & 0x0000'FF00) | (_Val >> 24);
+    } else {
+        return _byteswap_ulong(_Val);
+    }
+}
+
+_NODISCARD constexpr unsigned long long _Byteswap_uint64(const unsigned long long _Val) noexcept {
+    if (_STD is_constant_evaluated()) {
+        return (_Val << 56) | ((_Val << 40) & 0x00FF'0000'0000'0000) | ((_Val << 24) & 0x0000'FF00'0000'0000)
+             | ((_Val << 8) & 0x0000'00FF'0000'0000) | ((_Val >> 8) & 0x0000'0000'FF00'0000)
+             | ((_Val >> 24) & 0x0000'0000'00FF'0000) | ((_Val >> 40) & 0x0000'0000'0000'FF00) | (_Val >> 56);
+    } else {
+        return _byteswap_uint64(_Val);
+    }
+}
+
+_EXPORT_STD template <class _Ty, enable_if_t<is_integral_v<_Ty>, int> = 0>

```

Comment:
Out of curiosity:
Is there any special reason that in the `C++23` mode an ugly `enable_if_t` is used instead of beautiful concepts?

Reply:
Unfortunately, yes. See #395.